Set the right PATH for ARM compiler and build tests

ppigazzini opened the pull request:
This GitHub upgrade broke our ARM workflows because
it changed the value of the ANDROID_NDK_HOME variable
referenced in our PATH
https://github.com/actions/runner-images/issues/5879

Disservin commented:
Cool! ;)

vondele commented:
nice, I'll merge that. That probably should be a good starting point for generating also the binaries as part of the CI, but that's an independent thing.

ppigazzini commented:
> nice, I'll merge that. That probably should be a good starting point for generating also the binaries as part of the CI, but that's an independent thing.

This should `make build` the ARM binaries:
https://github.com/official-stockfish/Stockfish/compare/master...ppigazzini:Stockfish:arm_build?expand=1
https://github.com/ppigazzini/Stockfish/actions/runs/3652119062

Locally I'm able to `make profile-build` with NDK setting `WINE_PATH=qemu-aarch64`, unfortunately on GitHub I got an unexpected linker error during the bench:
https://github.com/official-stockfish/Stockfish/compare/master...ppigazzini:Stockfish:arm_profile_build?expand=1

```
Step 2/4. Running benchmark for pgo-build ...
qemu-aarch64 ./stockfish bench 2>&1 | tail -n 4
/system/bin/linker64: No such file or directory
```

ppigazzini commented:
This should `make profile-build` the ARM binaries:
https://github.com/official-stockfish/Stockfish/compare/master...ppigazzini:Stockfish:arm_profile_build?expand=1
https://github.com/ppigazzini/Stockfish/actions/runs/3652520216

I simply forgotten to set the static build :)

vondele commented:
I looked at your branch, I think that would work. Any chance one can rename stockfish-ubuntu-20.04-armv7-neon  into something like stockfish-android-armv7-neon (to avoid the confusion with ubuntu in the artifact name?)

ppigazzini commented:
Updated using Android/android in few places:
https://github.com/official-stockfish/Stockfish/compare/master...ppigazzini:Stockfish:arm_profile_build?expand=1
https://github.com/ppigazzini/Stockfish/actions/runs/3652773915

I'm not an expert of the GitHub yaml syntax, so I added a new file for the ARM artifacts. I kept some unutilized variables if someone want to merge the two artifacts files.
